// ng5-time-picker/src/app/components/w-mat-timepicker/w-mat-timepicker.module.js
import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MatButtonModule, MatButtonToggleModule, MatDialogModule, MatIconModule, MatInputModule, MatSelectModule, MatToolbarModule, MatFormFieldModule, MatCardModule } from '@angular/material';
import { WMatTimePickerComponent } from './w-mat-timepicker.component';
import { WTimeDialogComponent } from '../w-time-dialog/w-time-dialog.component';
import { WClockComponent } from '../w-clock/w-clock.component';
import { WTimeComponent } from '../w-time/w-time.component';
var Ng5TimePickerModule = /** @class */ (function () {
    function Ng5TimePickerModule() {
    }
    Ng5TimePickerModule.forRoot = function () {
        return {
            ngModule: Ng5TimePickerModule
        };
    };
    Ng5TimePickerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        MatButtonModule,
                        MatButtonToggleModule,
                        MatDialogModule,
                        MatFormFieldModule,
                        MatIconModule,
                        MatInputModule,
                        MatSelectModule,
                        MatToolbarModule,
                        FlexLayoutModule,
                    ],
                    exports: [
                        WMatTimePickerComponent,
                        WTimeDialogComponent,
                        WClockComponent,
                        WTimeComponent,
                    ],
                    declarations: [
                        WMatTimePickerComponent,
                        WTimeDialogComponent,
                        WClockComponent,
                        WTimeComponent,
                    ],
                    entryComponents: [
                        WMatTimePickerComponent,
                        WTimeDialogComponent,
                        WClockComponent,
                        WTimeComponent,
                    ]
                },] },
    ];
    /** @nocollapse */
    Ng5TimePickerModule.ctorParameters = function () { return []; };
    return Ng5TimePickerModule;
}());
export { Ng5TimePickerModule };
